### DISCUSSION TITLE
How Much VRAM Required to Fine Tune XTTS V2

### DISCUSSION THREAD

**Discussion by Priyanshu-hawk** [coqui-ai/TTS#3268] [General Q&A]
I am using RTX 3060 12GB VRAM.

I am trying fine tune a xtts v2 with its default settings.
xtts_v2 training file - https://github.com/coqui-ai/TTS/blob/dev/recipes/ljspeech/xtts_v2/train_gpt_xtts.py
<br>

**I got Out Of Memory Error with:**
```
BATCH_SIZE = 3 
GRAD_ACUMM_STEPS = 84
```

**I got Out Of Memory Error with:**
```
BATCH_SIZE = 2
GRAD_ACUMM_STEPS = 126
```

**It is Working with**
```
BATCH_SIZE = 1 
GRAD_ACUMM_STEPS = 252
```
<br>

**Comment by insomnia777** [comment]
![2023-11-20-004945_1491x81_scrot](https://github.com/coqui-ai/TTS/assets/88548335/c3f37333-d285-473e-b5f1-4020ed839d0b)
14Gb

**Comment by erogol** [comment]
Did you try float16?

**Reply by Priyanshu-hawk** [reply]
I think it is default to fp16. should i go for bf16?

![image](https://github.com/coqui-ai/TTS/assets/76026651/ab275a58-9012-488c-8459-924761951849)

**Comment by Tessory** [comment]
@Priyanshu-hawk I recommend to update nvidia driver. There new feature "CUDA – Sysmem Fallback Policy” in nvidia settings. I now on 3070 8Gb easilly finetuning. VRAM using show up to 17Gb in Task Manager. It's use half of RAM for VRAM. Really helped me a lot. On old nvidia driver I also previously got error pytorch like you

**Reply by Priyanshu-hawk** [reply]
Hello @Tessory, i'm using linux mint as my OS.
I'm currently on Nvidia Driver **525.147.05** . Should i update to **535** ?
Or **CUDA – Sysmem Fallback Policy** is some sort of setting which i need to change?

![image](https://github.com/coqui-ai/TTS/assets/76026651/bd738f83-2399-4d05-9f22-36bd84ac765c)

I'm using
**CUDA - 12.1
CuDnn - 8.9.2** 

I saw on https://developer.nvidia.com/rdp/cudnn-archive that **Cudnn - 8.9.2** releases on June 1st 2023

And one more cudnn is available which is **8.9.5** and releases on October 27th 2023.

So should i go from Cudnn **8.9.2** to **8.9.6**?

**Reply by Tessory** [reply]
Sorry, I don't know about linux drivers. I use it on windows. For windows that new feature just appear in latest november drivers. You can find CUDA – Sysmem Fallback Policy under ‘Manage 3D settings’ in the Nvidia driver settings. It's really did magic for me. XTTS fine-turn training now possible on my 8Gb VRAM card. Just google "CUDA – Sysmem Fallback Policy". Now a lot of people talking about it on many forums.

**Reply by Priyanshu-hawk** [reply]
Ok I'll try to do it for linux
Thanks.

**Comment by HobisPL** [comment]
Hey, I'm trying to fine-tune the XTTS_v2 model, and even setting batch_size = 1 and other configurations, I'm getting an out-of-memory error due to lack of VRAM. I have 8GB of VRAM, and I trained the model for Tortoise without any problems. It seems to me that the code might have been incorrectly rewritten...

**Reply by Tessory** [reply]
If you training on Windows, then try to update nVidia driver to latest. There new feature CUDA – Sysmem Fallback Policy under ‘Manage 3D settings’ in the Nvidia driver settings. With that feature GPU now can use CPU RAM. I also got that pytorch error like you, but with new drivers all fine-turned training fine. My GPU also 8Gb VRAM. Task manager show that GPU used up to 17Gb (VRAM+RAM). Really impressive. I'm ready to shake Jensen Huang hand for such a cool feature!

**Reply by HobisPL** [reply]
Indeed, I updated the driver and now the training is working; previously, it was doing 3 steps, and now it completed 1 epoch. Thanks a lot!

**Comment by HobisPL** [comment]
> Did you try float16?

where can I change?
